src/app/utils/transaction.ts

```diff
@@ -0,0 +1,28 @@
+import { ConsensusTxMethod, Transaction } from '../../oasis-nexus/api'
+
+export const getConsensusTransactionToAddress = (transaction: Transaction) => {
+  switch (transaction.method) {
+    case ConsensusTxMethod.stakingAddEscrow:
+      return transaction.body?.address
+    case ConsensusTxMethod.stakingAllow:
+      return transaction.body?.beneficiary
+    case ConsensusTxMethod.stakingReclaimEscrow:
+      return transaction.body?.account
+    case ConsensusTxMethod.stakingTransfer:
+      return transaction.body?.to
+    default:
+      return undefined
+  }
+}
+
+export const getConsensusTransactionAmount = (transaction: Transaction) => {
```

Comment:
Maybe this should be used in api.ts and extend Transaction definition; so various lists and details views can simply use it